src/js/components/Select/SelectContainer.js

```diff
@@ -131,31 +153,35 @@ class SelectContainer extends Component {
       >
         <Box
           id={id ? `${id}__select-drop` : undefined}
-          background={dropBackground}
         >
-          {onSearch ? (
-            <Box pad='xsmall'>
-              <TextInput
-                focusIndicator={true}
+          {onSearch && (
+            <Box pad={!customSearchInput ? 'xsmall' : undefined} flex={false}>
+              <SelectTextInput
+                focusIndicator={!customSearchInput}
                 size='small'
-                ref={(ref) => { this.searchRef = ref; }}
+                ref={this.searchRef}
                 type='search'
                 value={search}
                 placeholder={searchPlaceholder}
                 onInput={this.onInput}
               />
             </Box>
-          ) : undefined}
-
+          )}
           <Box
-            flex={false}
+            flex={true}
             role='menubar'
             tabIndex='-1'
-            ref={(ref) => { this.selectRef = ref; }}
+            ref={this.selectRef}
+            style={{
```

Comment:
Should these be in a styled component instead of inline styles?

Reply:
agreed. let me change that.